lib/semanage.c

```diff
@@ -23,27 +23,23 @@
 
 #include "attr.h"
 #include "prototypes.h"
-
 #include "shadowlog_internal.h"
+#include "string/sprintf/aprintf.h"
 
 
 format_attr(printf, 3, 4)
 static void semanage_error_callback (MAYBE_UNUSED void *varg,
                                      semanage_handle_t *handle,
                                      const char *fmt, ...)
 {
-	int ret;
 	char * message = NULL;
 	va_list ap;
 
-
 	va_start (ap, fmt);
-	ret = vasprintf (&message, fmt, ap);
+	message = vaprintf(fmt, ap);
```

Comment:
Why not use aprintf here and drop the va_list?

Reply:
semanage_error_callback() is a variadic function.  You need to use va_list to reference its argument list.  It's impossible to do use aprintf() in this code, unless I'm not understanding what you're suggesting.

Reply:
Oh.. right